lab_1/main.py

```diff
@@ -30,7 +52,14 @@ def calculate_frequencies(tokens: list) -> dict or None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
-    pass
+    if not isinstance(tokens, list):
+        return None
+    for i in tokens:
+        if not i:
+            return None
+
+    freq_dict = {i: tokens.count(i) for i in tokens}
```

Comment:
can you explain how this works? in writing, it's obligatory

Reply:
If you don't mind, I'm going to answer in russian just to avoid misunderstandings.
В данной функции для создания частотного словаря я использовала генератор словарей, который в соответствие токену-ключу(i) ставит количество его вхождений в список токенов(tokens.count(i))